drivers/storage/portworx/component/securitycontextconstraints.go

```diff
@@ -85,18 +85,28 @@ func (s *scc) Reconcile(cluster *opcorev1.StorageCluster) error {
 	// Note SCC does not belong to namespace hence we don't set owner reference to StorageCluster.
 	// By design cross-namespace owner reference is invalid.
 	for _, scc := range s.getSCCs(cluster) {
+		existingScc := ocp_secv1.SecurityContextConstraints{}
 		err := s.k8sClient.Get(context.TODO(),
 			types.NamespacedName{
 				Name: scc.Name,
 			},
-			&ocp_secv1.SecurityContextConstraints{})
+			&existingScc)
 
 		if errors.IsNotFound(err) {
 			err = s.k8sClient.Create(context.TODO(), &scc)
-		}
-
-		if err != nil {
-			return fmt.Errorf("failed to create %s security context constraints: %s", scc.Name, err)
+			if err != nil {
+				return fmt.Errorf("failed to create %s security context constraints: %s", scc.Name, err)
+			}
+		} else if err == nil {
+			if !reflect.DeepEqual(scc, existingScc) {
```

Comment:
actually this will never be equal, i just checked on my cluster, the scc has these fields which are changed after it;s deployed:

metadata:
  creationTimestamp: "2023-03-07T01:30:11Z"
  generation: 1
  name: portworx
  resourceVersion: "6870447"
  uid: 4feeeb20-9d29-4c7e-b1ad-aac7af8ab883

Could you make this API as a public API in util, and then call this API to cleanup the object before comparing https://github.com/libopenstorage/operator/blob/master/pkg/dryrun/dryrun.go#L425

also write a log before updating object and verify that it's only updated once.

Reply:
Resolved in rev-2